openc3/spec/interfaces/protocols/slip_protocol_spec.rb

```diff
@@ -0,0 +1,266 @@
+        expect {
+          @interface.add_protocol(SlipProtocol, [nil, true, true, nil],:READ_WRITE)
+        }.to raise_error(/write_enable_escaping must be true or false/)
+        expect {
+          @interface.add_protocol(SlipProtocol, [nil, true, true, nil, "5.1234"],:READ_WRITE)
+        }.to raise_error(/invalid value for Integer/)
+        expect {
+          @interface.add_protocol(SlipProtocol, [nil, true, true, nil, "0xC0", "5.1234"],:READ_WRITE)
+        }.to raise_error(/invalid value for Integer/)
+        expect {
+          @interface.add_protocol(SlipProtocol, [nil, true, true, nil, "0xC0", "0xDB", "5.1234"],:READ_WRITE)
+        }.to raise_error(/invalid value for Integer/)
+        expect {
+          @interface.add_protocol(SlipProtocol, [nil, true, true, nil, "0xC0", "0xDB", "0xDC", "5.1234"],:READ_WRITE)
+        }.to raise_error(/invalid value for Integer/)
+      end
+
+      it "handles proper params" do
+        protocol = @interface.add_protocol(SlipProtocol, ["0xC0", "false", "true", "false", "0xC0", "0xDB", "0xDC", "0xDD"],:READ_WRITE)
+        expect(protocol.instance_variable_get(:@start_char)).to eq "\xC0"
+        expect(protocol.instance_variable_get(:@read_strip_characters)).to eq false
+        expect(protocol.instance_variable_get(:@read_enable_escaping)).to eq true
+        expect(protocol.instance_variable_get(:@write_enable_escaping)).to eq false
+        expect(protocol.instance_variable_get(:@end_char)).to eq "\xC0"
+        expect(protocol.instance_variable_get(:@esc_char)).to eq "\xDB"
+        expect(protocol.instance_variable_get(:@esc_end_char)).to eq "\xDC"
+        expect(protocol.instance_variable_get(:@esc_esc_char)).to eq "\xDD"
+        expect(protocol.instance_variable_get(:@replace_end)).to eq "\xDB\xDC"
+        expect(protocol.instance_variable_get(:@replace_esc)).to eq "\xDB\xDD"
+      end
+    end
+
+    describe "read" do
+      it "handles multiple reads" do
+        $index = 0
+        class MultiTerminatedSlipStream < SlipStream
+          def read
+            case $index
+            when 0
+              $index += 1
+              "\x01\x02"
+            when 1
+              $index += 1
+              "\xC0"
+            end
+          end
+        end
+
+        @interface.instance_variable_set(:@stream, MultiTerminatedSlipStream.new)
+        @interface.add_protocol(SlipProtocol, [], :READ_WRITE)
+        packet = @interface.read
+        expect(packet.buffer).to eql("\x01\x02")
+      end
+
+      it "handles multiple reads" do
+        $index = 0
+        class MultiTerminatedSlipStream < SlipStream
+          def read
+            case $index
+            when 0
+              $index += 1
+              "\xC0"
+            when 1
+              $index += 1
+              "\x01\x02"
+            when 2
+              $index += 1
+              "\xC0"
+            when 3
+              $index += 1
+              "\xC0\x03\x04"
```

Comment:
So it simply throws away multiple continuous termination characters. Makes sense.